sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -784,6 +795,19 @@ case class Aggregate(
     }
   }
 
+  private def expandGroupingReferences(e: Expression): Expression = {
+    e match {
+      case _: AggregateExpression => e
+      case _ if PythonUDF.isGroupedAggPandasUDF(e) => e
```

Comment:
nit: now we can write `case _ if AggregateExpression.isAggregtate(e) => e`

Reply:
Fixed in https://github.com/apache/spark/pull/31913/commits/cff9b9ab289412529728749001ded9550f14e66e